ctt/model.py

```diff
@@ -0,0 +1,158 @@
+import dataclasses
+
+import oci.model
+import ocm
+
+
+@dataclasses.dataclass
+class ReplicationResourceOptions:
+    extra_tags: list[str] = dataclasses.field(default_factory=list)
+    remove_files: list[str] = dataclasses.field(default_factory=list)
+    digest: str | None = None
+    reference_by_digest: bool = False
+    retain_symbolic_tag: bool = False
+    convert_to_relative_ref: bool = False
+
+
+@dataclasses.dataclass(kw_only=True)
+class ReplicationResourceElement(ReplicationResourceOptions):
+    source: ocm.Resource
+    target: ocm.Resource
+    component_id: ocm.ComponentIdentity
+
+    @property
+    def src_ref(self) -> oci.model.OciImageReference:
+        if self.source.access.type is ocm.AccessType.OCI_REGISTRY:
+            return oci.model.OciImageReference(self.source.access.imageReference)
+        elif self.source.access.type is ocm.AccessType.RELATIVE_OCI_REFERENCE:
+            return oci.model.OciImageReference(self.source.access.reference, normalise=False)
+        else:
+            raise ValueError(self.source.access.type)
+
+    @property
+    def tgt_ref(self) -> oci.model.OciImageReference:
+        if self.target.access.type is ocm.AccessType.OCI_REGISTRY:
+            return oci.model.OciImageReference(self.target.access.imageReference)
+        elif self.target.access.type is ocm.AccessType.RELATIVE_OCI_REFERENCE:
+            return oci.model.OciImageReference(self.target.access.reference, normalise=False)
+        else:
+            raise ValueError(self.source.access.type)
+
+    @property
+    def preliminary_tgt_ref(self) -> oci.model.OciImageReference:
+        '''
+        In case the resource does not exist in the target yet, we don't know the digest for sure
+        (for example, there might be filtering). To show a reasonable output in the replication plan
+        nonetheless, use a placeholder digest in case no digest can be determined to show that the
+        replicated resource will contain a digest tag.
+        '''
+        if not self.reference_by_digest:
+            return self.tgt_ref
+
+        tgt_ref = self.tgt_ref
+
+        if self.digest:
+            digest = self.digest
+        elif tgt_ref.has_digest_tag:
+            digest = self.tgt_ref.tag
+        else:
+            digest = 'sha256:<placeholder-digest>'
```

Comment:
nit: should we make this slightly more self-describing, like `"<placeholder-if-you-see-me-this-indicates-a-bug>"`?